pages/sponsorship.js

```diff
@@ -0,0 +1,170 @@
+import Card, { CardHeader, CardTitle, CardContent, CardButtonContainer, CardDiv, TableDiv } from '../components/card'
+import styled from 'styled-components'
+import Button from '../components/button'
+import React, { useState } from 'react'
+import Modal from 'react-modal'
+import { EDIT, NEW } from '../constants'
+
+
+const customStyles = {
+    content : {
+      top                   : '50%',
+      left                  : '50%',
+      right                 : 'auto',
+      bottom                : 'auto',
+      marginRight           : '-50%',
+      transform             : 'translate(-50%, -50%)'
+    }
+  };
+
+
+// TODO: move font family higher, reduce need to redefine
+const HeaderCard = styled.div`
+    padding: 26px 28px;
+    font-weight: bold;
+    font-family: Apercu Pro; 
+    font-size: 16px;
+    border-bottom: 2px solid #EDEDED;
+
+`
+const Text = styled.text`
+    padding-right: 110px;
+`
+
+
+
+// map CardDiv to a function
+// async firebase api first 
+
+class SponsorshipPage extends React.Component {
+    
+    constructor(){
+        super();
+        this.state = {
+            sponsortest: [
+                {"name": "Tencent",
+                "link":"www.microsoft.com",
+                "image": "path to photo",
+                "lastmod": "today lol"
+                 },
+                {"name": "Apple",
+                "link":"www.apple.com",
+                "image": "yeeters",
+                "lastmod": "today lol"
+                },  
+                {"name": "Balenciaga",
+                "link":"www.ssense.com",
+                "image": "koko",
+                "lastmod": "lol"
+            }], 
+            showEditWindow: false,
+            "name": "",
+            "link": "",
+            "image": "",
+            "lastmod": ""
+            
+        }
+       
+        
+    }
+
+    
+    handleEdit = (e) => {
+        e.preventDefault();
+        //TODO: pop-up window 
+        this.setState({showEditWindow: true})
+    };
```

Comment:
A bit more modern and cleaner to use this syntax for your buttons
`<Button onClick={() => this.handleEdit()}>`

Then you can do 
```suggestion
    handleEdit = () => {
        //TODO: pop-up window 
        this.setState({showEditWindow: true})
    };
```